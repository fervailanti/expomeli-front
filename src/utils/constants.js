export const API_BASE_URL = 'https://expomeli.herokuapp.com/api'

export const PRICE_FORMAT = 'de-DE'

export const CONDITIONS = {
  new: 'Nuevo',
  used: 'Usado',
  default: 'Desconocido'
}

export const CURRENCIES = {
  ARS: '$',
  USD: 'U$S',
  PEN: 'S/',
  default: '$'
}
