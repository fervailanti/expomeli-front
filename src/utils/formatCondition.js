export const formatCondition = (condition) => {
  const options = {
    new: 'Nuevo',
    used: 'Usado',
    default: 'Desconocido'
  }
  return options[condition] || options.default
}
