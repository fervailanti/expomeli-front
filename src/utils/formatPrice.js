export const formatPrice = (price) => {
  const { currency, amount, decimals } = price
  const options = {
    ARS: '$',
    USD: 'U$S',
    PEN: 'S/',
    default: '$'
  }
  const formattedCurrency = options[currency] || options.default
  const formattedAmount = Intl.NumberFormat('de-DE').format(amount)
  const fixedPrice = decimals ? `${formattedAmount},${decimals}` : formattedAmount
  return `${formattedCurrency} ${fixedPrice}`
}
