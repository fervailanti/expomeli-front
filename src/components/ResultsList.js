import React from 'react'
import ResultItem from './ResultItem'
import { formatPrice } from '../utils/formatPrice'
import { WithLoader } from './WithLoader'
import { formatCondition } from '../utils/formatCondition'
import Breadcrumb from './Breadcrumb'
import EmptyState from './EmptyState'

const ResultsList = ({ results, onItem }) => {
  const { categories, items } = results
  const content = {
    empty: () => <EmptyState />,
    list: () =>
      items.map((item, index) => {
        const action = () => onItem(item.id)
        return (
          <ResultItem
            key={index}
            image={item.picture}
            title={item.title}
            freeShipping={item.free_shipping}
            condition={formatCondition(item.condition)}
            price={formatPrice(item.price)}
            action={action}
          />
        )
      })
  }
  return (
    <>
      {categories.length > 0 && <Breadcrumb categories={categories} />}
      <div className="results-list">{items.length ? content.list() : content.empty()}</div>
    </>
  )
}

export default WithLoader(ResultsList)
