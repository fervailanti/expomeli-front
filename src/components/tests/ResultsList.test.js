import React from 'react'
import ResultsList from '../ResultsList'
import { render } from '@testing-library/react'
import { autor } from '../../utils/constants'

const itemsMock = {
  autor,
  categories: ['Apple', 'iPhone'],
  items: Array(4)
    .fill()
    .map((_, i) => ({
      id: `TEST${i}`,
      title: `iPhone 11 Pro ${i}`,
      price: {
        currency: 'ARS',
        amount: 34140 + i,
        decimals: 20 + i
      },
      picture: 'https://picsum.photos/90',
      condition: 'new',
      free_shipping: true
    }))
}

it('ResultsList component should render correctly', () => {
  const tree = render(<ResultsList results={itemsMock} onItem={() => null} />)
  expect(tree.container).toMatchSnapshot()
})
