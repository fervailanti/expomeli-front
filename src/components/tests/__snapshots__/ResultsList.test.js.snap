// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultsList component should render correctly 1`] = `
<div>
  <div
    class="flex flex-wrap items-center mb-3 lg:mb-5"
  >
    <p
      class="font-thin text-sm lg:text-base"
    >
      Apple
    </p>
    <img
      alt="arrow-icon"
      class="h-3 mx-2 last:hidden"
      src="[object Object]"
    />
    <p
      class="font-thin text-sm lg:text-base"
    >
      iPhone
    </p>
    <img
      alt="arrow-icon"
      class="h-3 mx-2 last:hidden"
      src="[object Object]"
    />
  </div>
  <div
    class="results-list"
  >
    <div
      class="item"
    >
      <img
        alt="product"
        class="img"
        src="https://picsum.photos/90"
      />
      <div
        class="main"
      >
        <p
          class="condition"
        >
          Nuevo
        </p>
        <p
          class="title"
        >
          iPhone 11 Pro 0
        </p>
        <p
          class="price"
        >
          $ 34.140,20
        </p>
        <p
          class="tag"
        >
          Envío gratis
        </p>
      </div>
    </div>
    <div
      class="item"
    >
      <img
        alt="product"
        class="img"
        src="https://picsum.photos/90"
      />
      <div
        class="main"
      >
        <p
          class="condition"
        >
          Nuevo
        </p>
        <p
          class="title"
        >
          iPhone 11 Pro 1
        </p>
        <p
          class="price"
        >
          $ 34.141,21
        </p>
        <p
          class="tag"
        >
          Envío gratis
        </p>
      </div>
    </div>
    <div
      class="item"
    >
      <img
        alt="product"
        class="img"
        src="https://picsum.photos/90"
      />
      <div
        class="main"
      >
        <p
          class="condition"
        >
          Nuevo
        </p>
        <p
          class="title"
        >
          iPhone 11 Pro 2
        </p>
        <p
          class="price"
        >
          $ 34.142,22
        </p>
        <p
          class="tag"
        >
          Envío gratis
        </p>
      </div>
    </div>
    <div
      class="item"
    >
      <img
        alt="product"
        class="img"
        src="https://picsum.photos/90"
      />
      <div
        class="main"
      >
        <p
          class="condition"
        >
          Nuevo
        </p>
        <p
          class="title"
        >
          iPhone 11 Pro 3
        </p>
        <p
          class="price"
        >
          $ 34.143,23
        </p>
        <p
          class="tag"
        >
          Envío gratis
        </p>
      </div>
    </div>
  </div>
</div>
`;
